### PR TITLE
fix: warn if the patch version is passed in kubernetes version flag

### LINKED ISSUE #414: wrong version of Kubernetes deployed when k8s distro set
### What happened?

The command I used:

vcluster create --kubernetes-version v1.21.8 vcluster-1 -n host-namespace-1 --distro k8s

The version I got:

Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5", GitCommit:"aea7bbadd2fc0cd689de94a54e5b7b758869d691", GitTreeState:"clean", BuildDate:"2021-09-15T21:04:16Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}

### What did you expect to happen?

Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.**8**", 

### How can we reproduce it (as minimally and precisely as possible)?

command I used above

### Anything else we need to know?

_No response_

### Host cluster Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Host cluster Kubernetes distribution

<details>

```
# Write here
```

</details>

### vlcuster version

<details>

```console
$ vcluster --version
# paste output here
```

</details>

### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
# Write here
```

</details>

### OS and Arch

<details>

```
OS: 
Arch:
```
</details>

### PR DESCRIPTION
**Pull request by pratikjagrut**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind bugfix

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #414

### CODE REVIEW

File: pkg/helm/values/k3s.go [hunk 1]
```diff
@@ -136,15 +138,16 @@ isolation:
 	return values, nil
 }
 
-func ParseKubernetesVersionInfo(versionStr string) (*version.Info, error) {
-	if versionStr[0] == 'v' {
-		versionStr = versionStr[1:]
```
Comment (FabianKramm) [inline pkg/helm/values/k3s.go:141]: @pratikjagrut does `--kubernetes-version v1.20` and `--kubernetes-version 1.20` still work?

**Review by FabianKramm — COMMENTED** [review]
@pratikjagrut thanks for the PR! Just one small question, rest looks good!

File: pkg/helm/values/k3s.go (hunk 1 above)
Comment (pratikjagrut) [reply pkg/helm/values/k3s.go:141]: v1.20, but just 1.20 will not work because it is not in a semvar format.

Comment (FabianKramm) [reply pkg/helm/values/k3s.go]: @pratikjagrut I think we should allow 1.20 as well, as its easy to do for us and we definitely know what the user wants to do. Helm for example also allows this for chart versions even though they require semver for it as well.

Comment (pratikjagrut) [reply pkg/helm/values/k3s.go]: I’ve already updated this to allow 1.20 too.

**Review by FabianKramm — APPROVED** [review]
@pratikjagrut thanks for the PR! LGTM!

File: pkg/helm/values/k3s.go (hunk 1 above)
Comment (FabianKramm) [reply pkg/helm/values/k3s.go]: Ah didn't see, my bad! I'll merge it!

**Pull request merged**